.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,164 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Workflow for all benchmark-related jobs. It is designed to be called from
+# the main workflow ci.yml. The concurrency of this workflow is controlled by
+# the caller's job.
+
+name: Benchmarks
+
+on:
+  workflow_call:
+    inputs:
+      runner-group:
+        required: true
+        type: string
+      runner-env:
+        required: true
+        type: string
+      e2e-test-artifacts-dir:
+        required: true
+        type: string
+      e2e-test-artifacts-gcs-artifact-dir:
+        required: true
+        type: string
+      benchmark-presets:
+        required: true
+        type: string
+
+env:
+  # This duplicates the variable from ci.yml. The variable needs to be in env
+  # instead of the outputs of setup because it contains the run attempt and we
+  # want that to be the current attempt, not whatever attempt the setup step
+  # last ran in. It therefore can't be passed in via inputs because the env
+  # context isn't available there.
+  GCS_DIR: gs://iree-github-actions-${{ github.event_name == 'pull_request' && 'presubmit' || 'postsubmit' }}-artifacts/${{ github.run_id }}/${{ github.run_attempt }}
+
+jobs:
+  export_benchmark_config:
+    runs-on:
```

Comment:
Github hosted runners can be used here but I couldn't because Github hosted runners because those runners don't have write access to GCS.